wpengine/atlas-blueprint-portfolio
# [MERL-222] Finalize blueprint zip

# Description

Depends on #65 

Updates a few things in the zip:
- Uses the new `with_front` option on the Project model to make project URIs look like `/project/[projectSlug]`* rather than `/posts/project/[projectSlug]`.
- Updates the Portfolio menu item URL to `/project`*
- Adds a new Footer navigation menu and assigns it to the Footer menu location. I thought this might help demonstrate that the footer can be different than the main navigation.
- Removes the default "Sample Page" that snuck in at some point.

*Ideally we would have been able to keep `/projects` (plural), but custom post types to use the (singular) post type slug in the URI by default. That means we are stuck with `/project` unless ACM decides to modify this behavior. There were a few options discussed in Slack:

1. **Don't use the `with_front` feature.**

    Leaves things as they are, meaning project URIs from WordPress would not match the frontend. The search results page needs URIs to be handled consistently across post types, so this option is out.

3. **Use the `with_front` feature and rename the `src/pages/projects` directory to `src/pages/project`.** The "Portfolio" page would be accessible from `/project` and each individual project would be at `/project/[projectSlug]`.

    I went with this option. The `/project/[projectSlug]` URIs still make logical sense and we already have the precedent of `/category/[categorySlug]`. The tradeoff is that `/project` isn't a great URI for an index listing multiple projects. We could change this with option 3...

4. **Same as option 2, but we also move `src/pages/project/index.js` to `src/pages/projects.js`.** The "Portfolio" page would be accessible from `/projects` and each individual project would be at `/project/[projectSlug]`.

    This would result in a better index URL for site visitors, but require a slightly less desirable project structure. I decided not to make this change considering the blueprint's role as a sample/learning tool rather than an actual production site, but I'm open to being convinced otherwise 😄 
    
# Testing

1. Make sure you create a fresh WP Install. Do not use WP Reset plugin or anything similar as it may not work.

2. Open a shell.

3. Import the blueprint:

```
wp acm blueprint import https://github.com/wpengine/atlas-blueprint-portfolio/raw/merl-222-finalize-zip/acm-blueprint.zip
```
